Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5369,8 +5383,11 @@ void WebPageProxy::scalePage(double scale, const IntPoint& origin, CompletionHan
     }
 
     send(Messages::WebPage::DidScalePage(scale, origin));
-    forEachWebContentProcess([&] (auto& process, auto pageID) {
-        if (&process == &legacyMainFrameProcess())
+    forEachWebContentProcess([weakThis = WeakPtr { *this }, scale] (auto& process, auto pageID) {
```

Comment:
This is not asynchronous, it doesn't make sense to capture a weak ptr. Just capture `page = Ref { *this }`